src/main/java/teammates/ui/webapi/request/StudentsEnrollRequest.java

```diff
@@ -0,0 +1,107 @@
+package teammates.ui.webapi.request;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * The request for enrolling a list of students.
+ */
+public class StudentsEnrollRequest extends BasicRequest {
+
+    private static final String ERROR_MESSAGE_SAME_TEAM_IN_DIFFERENT_SECTION =
+            "Team \"%s\" is detected in both Section \"%s\" and Section \"%s\"."
+                    + " Please use different team names in different sections";
+    private static final String ERROR_MESSAGE_DUPLICATE_EMAIL =
+            "Error, duplicated email addresses detected in the input: %s";
+    private static final String ERROR_MESSAGE_TOO_MANY_ENROLLMENTS = "Please enroll at most 100 students each time.";
+    private static final String ERROR_MESSAGE_EMPTY_ENROLLMENT = "Please input at least one student detail.";
+
+    private List<StudentEnrollRequest> studentEnrollRequests;
+
+    public StudentsEnrollRequest(List<StudentEnrollRequest> studentEnrollRequests) {
+        this.studentEnrollRequests = studentEnrollRequests;
+    }
+
+    public List<StudentEnrollRequest> getStudentEnrollRequests() {
+        return studentEnrollRequests;
+    }
+
+    @Override
+    public void validate() {
+        assertTrue(!studentEnrollRequests.isEmpty(), ERROR_MESSAGE_EMPTY_ENROLLMENT);
+        assertTrue(studentEnrollRequests.size() <= 100, ERROR_MESSAGE_TOO_MANY_ENROLLMENTS);
+        for (StudentEnrollRequest request : studentEnrollRequests) {
+            request.validate();
+        }
+
+        Set<String> emails = new HashSet<>();
+        for (StudentEnrollRequest request : studentEnrollRequests) {
+            assertTrue(!emails.contains(request.getEmail()),
+                    String.format(ERROR_MESSAGE_DUPLICATE_EMAIL, request.getEmail()));
+            emails.add(request.getEmail());
+        }
+
+        Map<String, String> teamInSection = new HashMap<>();
```

Comment:
There is a problem of putting the team validation logic here.

The teams that are already enrolled is not valided together with the students to enrol.

Reply:
Yes, and I put that part of validation in frontend.

Should I merge two parts of validation together and remove the validation here? As I think this validation here is for the request itself, I keep both currently

Reply:
I see, for security reasons, I will put validation into backend